cpp/src/gandiva/precompiled/time_test.cc

```diff
@@ -766,6 +766,24 @@ TEST(TestTime, TestCastTimestampToDate) {
   EXPECT_EQ(StringToTimestamp("2000-05-01 00:00:00"), out);
 }
 
+TEST(TestTime, TestNextDay) {
+  gdv_timestamp ts = StringToTimestamp("2021-11-08 10:20:34");
+  auto out = next_day(ts, "FR", 2);
+  EXPECT_EQ(StringToTimestamp("2021-11-12 00:00:00"), out);
+
+  ts = StringToTimestamp("2021-11-08 10:20:34");
+  out = next_day(ts, "FRI", 3);
+  EXPECT_EQ(StringToTimestamp("2021-11-12 00:00:00"), out);
+
+  ts = StringToTimestamp("2021-11-08 10:20:34");
+  out = next_day(ts, "FRIDAY", 6);
```

Comment:
What happens if the user defines an invalid day as parameter? For example: `next_day(ts, KASJLHSDHJ, 2)